crates/bevy_ecs/src/schedule_v3/schedule.rs

```diff
@@ -0,0 +1,1027 @@
+    world::World,
+};
+
+/// Resource that stores [`Schedule`]s mapped to [`ScheduleLabel`]s.
+#[derive(Default, Resource)]
+pub struct Schedules {
+    inner: HashMap<BoxedScheduleLabel, Schedule>,
+}
+
+impl Schedules {
+    /// Constructs an empty `Schedules` with zero initial capacity.
+    pub fn new() -> Self {
+        Self {
+            inner: HashMap::new(),
+        }
+    }
+
+    /// Inserts a labeled schedule into the map.
+    ///
+    /// If the map already had an entry for `label`, `schedule` is inserted,
+    /// and the old schedule is returned. Otherwise, `None` is returned.
+    pub fn insert(&mut self, label: impl ScheduleLabel, schedule: Schedule) -> Option<Schedule> {
+        let label = label.dyn_clone();
+        if self.inner.contains_key(&label) {
+            warn!("schedule with label {:?} already exists", label);
+        }
+        self.inner.insert(label, schedule)
+    }
+
+    /// Removes the `label` entry from the map, returning its schedule if one existed.
+    pub fn remove(&mut self, label: &dyn ScheduleLabel) -> Option<Schedule> {
+        let label = label.dyn_clone();
+        if !self.inner.contains_key(&label) {
+            warn!("schedule with label {:?} not found", label);
+        }
+        self.inner.remove(&label)
+    }
+
+    /// Returns a reference to the schedule associated with `label`, if it exists.
+    pub fn get(&self, label: &dyn ScheduleLabel) -> Option<&Schedule> {
+        let label = label.dyn_clone();
+        self.inner.get(&label)
+    }
+
+    /// Returns a mutable reference to the schedule associated with `label`, if it exists.
+    pub fn get_mut(&mut self, label: &dyn ScheduleLabel) -> Option<&mut Schedule> {
+        let label = label.dyn_clone();
+        self.inner.get_mut(&label)
+    }
+
+    /// Iterates all system change ticks and clamps any older than [`MAX_CHANGE_AGE`](crate::change_detection::MAX_CHANGE_AGE).
+    /// This prevents overflow and thus prevents false positives.
+    ///
+    /// **Note:** Does nothing if the [`World`] counter has not been incremented at least [`CHECK_TICK_THRESHOLD`](crate::change_detection::CHECK_TICK_THRESHOLD)
+    /// times since the previous pass.
+    pub(crate) fn check_change_ticks(&mut self, change_tick: u32) {
+        #[cfg(feature = "trace")]
+        let _all_span = info_span!("check stored schedule ticks").entered();
+        // label used when trace feature is enabled
+        #[allow(unused_variables)]
+        for (label, schedule) in self.inner.iter_mut() {
+            #[cfg(feature = "trace")]
+            let name = format!("{:?}", label);
+            #[cfg(feature = "trace")]
+            let _one_span = info_span!("check schedule ticks", name = &name).entered();
+            schedule.check_change_ticks(change_tick);
+        }
+    }
+}
+
+/// A collection of systems, and the metadata and executor needed to run them
+/// in a certain order under certain conditions.
+pub struct Schedule {
```

Comment:
is there a way for a tool like bevy_mod_debugdump to get the graph info and display the schedule graph?

Reply:
What info does it want? `SystemContainer` doesn't exist anymore, but we could maybe generate something on demand.

Reply:
I would start by simply adding getter methods for the `ScheduleGraph` and `SystemSchedule` fields.

Reply:
I can add methods to `Schedule` that return references to the vectors inside `SystemSchedule`. I don't want `SystemSchedule` or `SystemExecutor` to be `pub` until we flesh out an API for writing custom executors.

Reply:
SGTM, that seems like a nice start.